### ISSUE TITLE
CLIError: Error in "Loading schema for gatsbySchema": Error: Cannot use GraphQLSchema "[object GraphQLSchema]" from another module or realm.

### ISSUE THREAD

**Issue by insuusvenerati** [daugsbi/gatsby-plugin-codegen#12]
Is there any way I would be able to fix this issue? Or is it something related to deps of the plugin?

```
cuccess [gatsby-plugin-codegen] saved localSchemaFile: ./schema.json
success [gatsby-plugin-codegen] saved apollo config: apollo.config.js

 ERROR 

CLIError: Error in "Loading schema for gatsbySchema": Error: Cannot use GraphQLSchema "[object GraphQLSchema]" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.

https://yarnpkg.com/en/docs/selective-version-resolutions

Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior. The data from one
version used in the function from another could produce confusing and
spurious results.
    at Object.error (D:\Code\JS\GatsbyJS\datocms-Gatsby-portfolio-demo\node_modules\@oclif\errors\lib\index.js:26:15)
    at Generate.error (D:\Code\JS\GatsbyJS\datocms-Gatsby-portfolio-demo\node_modules\@oclif\command\lib\command.js:60:23)
    at OclifLoadingHandler.showError (D:\Code\JS\GatsbyJS\datocms-Gatsby-portfolio-demo\node_modules\apollo\lib\OclifLoadingHandler.js:28:22)
    at OclifLoadingHandler.handle (D:\Code\JS\GatsbyJS\datocms-Gatsby-portfolio-demo\node_modules\apollo\lib\OclifLoadingHandler.js:13:18)
    at async GraphQLClientProject.loadServiceSchema (D:\Code\JS\GatsbyJS\datocms-Gatsby-portfolio-demo\node_modules\apollo-language-server\lib\project\client.js:94:9)
    at async Promise.all (index 1) {
  oclif: { exit: 2 },
  code: undefined
}

⠸ onPostBootstrap
  × Loading Apollo Project
    → spurious results.
warn [gatsby-plugin-codegen] could not generate types from queries - Error initializing Apollo GraphQL project "gatsbySchema": Error: Error in "Loading schema for gatsbySchema": Error: Cannot use GraphQLSchema "[object
GraphQLSchema]" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.

https://yarnpkg.com/en/docs/selective-version-resolutions

Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior. The data from one
version used in the function from another could produce confusing and
success Build manifest and related icons - 0.334s
```

**Comment by daugsbi** [comment]
I've tried to fix it today. I just came up with a workaround yet. Just add the following to your package.json.

```JSON
"resolutions": {
    "graphql": "^14.6.0"
  }
```

Help would be appreciated to solve this issue. I can't fix it with resolutions, because they are only used in the root package. Is there any workaround

The general problem is that the apollo dependency is resolved to a newer graphql version which is not compatible with Gatsby's.   I need to somehow overwrite/downgrade this behavior.
